### PR TITLE
Validation for VK_NV_cooperative_matrix2

### PR DESCRIPTION
**Pull request by jeffbolznv**

### CODE REVIEW

File: layers/core_checks/cc_spirv.cpp [hunk 1]
```diff
@@ -555,172 +579,255 @@ bool CoreChecks::ValidateCooperativeMatrix(const spirv::Module &module_state, co
...
+                                         local_size_x, phys_dev_props_core11.subgroupSize);
+                    }
+                    if (m.scope == VK_SCOPE_WORKGROUP_KHR) {
+                        const auto workgroup_size = local_size_x * local_size_y * local_size_z;
+                        if (workgroup_size >
+                            phys_dev_ext_props.cooperative_matrix_props2_nv.cooperativeMatrixWorkgroupScopeMaxWorkgroupSize) {
+                            skip |= LogError(
+                                "VUID-VkPipelineShaderStageCreateInfo-module-10169", module_state.handle(), loc,
+                                "SPIR-V (compute stage) Total local workgroup size (%" PRIu32
+                                ") is larger than cooperativeMatrixWorkgroupScopeMaxWorkgroupSize (%" PRIu32 ")/.",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_spirv.cpp:618]: this seems like a typo

```suggestion
                                ") is larger than cooperativeMatrixWorkgroupScopeMaxWorkgroupSize (%" PRIu32 ").",
```

File: layers/core_checks/cc_spirv.cpp [hunk 2]
```diff
@@ -555,172 +579,255 @@ bool CoreChecks::ValidateCooperativeMatrix(const spirv::Module &module_state, co
...
                         valid = true;
                         break;
                     }
                 }
+                if (enabled_features.cooperativeMatrixFlexibleDimensions) {
+                    for (uint32_t i = 0; i < cooperative_matrix_flexible_dimensions_properties.size(); ++i) {
+                        const auto &property = cooperative_matrix_flexible_dimensions_properties[i];
+
+                        if (property.scope == VK_SCOPE_WORKGROUP_KHR &&
+                            local_size_x * local_size_y * local_size_z != property.workgroupInvocations) {
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_spirv.cpp:666]: I noticed you did `const auto workgroup_size = local_size_x * local_size_y * local_size_z;` above... thinking about it now, maybe better to do that once at the top of the function if multiple spots are using it

**Review by spencer-lunarg — APPROVED** [review]
LGTM - with few nits

File: layers/core_checks/cc_spirv.cpp (hunk 1 above)
Comment (jeffbolznv) [reply layers/core_checks/cc_spirv.cpp]: done

File: layers/core_checks/cc_spirv.cpp (hunk 2 above)
Comment (jeffbolznv) [reply layers/core_checks/cc_spirv.cpp]: done

**Comment by spencer-lunarg** [comment]
> CI Vulkan-ValidationLayers build # 17907 failed.

Unrelated X Server issue, internal CI is good!

**Pull request merged**